eth/executionclient/multi_client.go

```diff
@@ -0,0 +1,365 @@
+package executionclient
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/ethereum/go-ethereum"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/sourcegraph/conc/pool"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/ssvlabs/ssv/eth/contract"
+)
+
+var _ Provider = &MultiClient{}
+
+type MultiClient struct {
+	// optional
+	logger *zap.Logger
+	// followDistance defines an offset into the past from the head block such that the block
+	// at this offset will be considered as very likely finalized.
+	followDistance              uint64 // TODO: consider reading the finalized checkpoint from consensus layer
+	connectionTimeout           time.Duration
+	reconnectionInitialInterval time.Duration
+	reconnectionMaxInterval     time.Duration
+	healthInvalidationInterval  time.Duration
+	logBatchSize                uint64
+	syncDistanceTolerance       uint64
+
+	contractAddress ethcommon.Address
+	chainID         *big.Int
+	closed          chan struct{}
+
+	nodeAddrs          []string
+	clients            []SingleClientProvider
+	currentClientIndex atomic.Int64
```

Comment:
really not a big deal, but perhaps `Uint64` would describe an index better

Reply:
since indexing in go is an `int` by default which is int64 , I thought it'll be an easier cast than uint.